### DISCUSSION TITLE
How to pixelate GIFs using python?

### DISCUSSION THREAD

**Discussion by Lypheal** [python-pillow/Pillow#6208] [Q&A] [answered]
I've been searching for this since months now... How can I pixelate gif using python?

For example from this
![image](https://user-images.githubusercontent.com/99541630/163364748-e1a5de8f-449d-4edf-afbf-105b8946ec51.gif)
to this
![image](https://user-images.githubusercontent.com/99541630/163364804-abbb1fdb-746a-4591-9032-0eeaa59fba41.gif)

How is this possible with python? Please someone help me.

**Comment by radarhere** [comment, marked as answer]
Running this code over your original image,
```python
from PIL import Image, ImageSequence

def pixelate(frame):
    if frame.mode == "P":
        frame = frame.convert("RGB")
    original_size = frame.size
    # To create the pixelation effect,
    # shrink the image by a factor of 5
    frame = frame.reduce(5)
    # and then resize it back to normal size
    return frame.resize(original_size, Image.Resampling.NEAREST)

# Open the image
with Image.open("in.gif") as im:
    # Run the "pixelate" function over every frame
    frames = ImageSequence.all_frames(im, pixelate)
    # Save it to file
    frames[0].save("out.gif", save_all=True, append_images=frames[1:])
```
gives
![out](https://user-images.githubusercontent.com/3112309/163382696-69b535d4-b1bd-4d8b-931b-a17ad3b1594f.gif)

**Reply by Lypheal** [reply]
How can I run this in google colab?

**Reply by radarhere** [reply]
1. Upload your input GIF to "Files"
2. `!python3 -m pip install --upgrade Pillow`. By default, the Pillow version was 7.1.2 when I tried.
3. Restart the runtime so that Pillow is updated.
4. Run the code I pasted above.
5. See the output GIF in "Files"

**Reply by Lypheal** [reply]
Okay thanks!
Just 1 more question, I have 100s of gifs in a folder I want to pixelate, how do i loop this so that it pixelates all of them without me needing to know the no. of files in that folder?

**Reply by radarhere** [reply]
Here is code that will loop over all of the GIF files in the current directory, using [`glob.glob`](https://docs.python.org/3/library/glob.html#glob.glob). Alternatively, you could use [`os.listdir`](https://docs.python.org/3/library/os.html#os.listdir).
```python
import glob
for filename in glob.glob("*.gif"):
    print(filename)
```

**Reply by Lypheal** [reply]
Thanks man!
I'm gonna use os.listdir since I don't trust glob nor do I have any experience with it
